### ISSUE TITLE
Need adequate example (docs) or convenient interface for PolySynth options

### ISSUE THREAD

**Issue by anselanza** [Tonejs/Tone.js#966]
**Example or convenient interface for PolySynth options**
When using Typescript, I am finding it impossible to specify the correct type to pass down to React components for the "options" parameter when constructing `new Tone.PolySynth`.

A helpful example of exactly what type signature (with generics, etc.) to use would be useful. Alternatively, some kind of alias might be useful if the type definition ends up becoming very hard to use (lots of generics, etc.).

**Any alternatives you've considered**
I have tried `PolySynth<Synth<SynthOptions>>`, for example:
```
const synthOptions: PolySynth<Synth<SynthOptions>> = {
  detune: 0,
  portamento: 0,
  harmonicity: 17,
  volume: -20,
  oscillator: {
// ... etc.
```
This type is allowed to be passed into `new Tone.PolySynth (Tone.FMSynth, synthOptions)` without complaint, but the TS compiler will complain about the definition of `synthOptions` itself: 
```
Type 'number' is not assignable to type 'Param<"decibels">'.ts(2322)
Instrument.d.ts(27, 5): The expected type comes from property 'volume' which is declared here on type 'PolySynth<Synth<SynthOptions>>'
```
(and so on).

This also does not work:
```
type Options = ConstructorParameters<Tone.PolySynth>
```
... fails with:
```
Type 'PolySynth<Synth<SynthOptions>>' does not satisfy the constraint 'abstract new (...args: any) => any'.
  Type 'PolySynth<Synth<SynthOptions>>' provides no match for the signature 'new (...args: any): any'.ts(2344)
```

My only viable solution for now is to use `any`:
```
interface Props {
  synthOptions: any;
// ...
```

**Additional context**
In spite of the recommendations of https://github.com/Tonejs/Tone.js/wiki/Using-Tone.js-with-React-React-Typescript-or-Vue I am not able to type options for a PolySynth instance without errors.

**Comment by anselanza** [comment]
Right, so we get a little closer by using `FMSynthOptions` but still the example provided on the website would break if used as-is:
![Screenshot 2021-11-01 at 14 18 41](https://user-images.githubusercontent.com/3015222/139678048-8288184c-6672-47e9-857b-1e843ec71ac7.png)

Notice that I have to add properties such as `muted`, `volume` and `onstop` in a few places (which seems redundant)... and... we're still getting compiler complaints!

**Comment by tambien** [comment]
If you don't want to fill in **all** of the interface attributes, set your type to be `const synthOptions: Partial<FMSynthOptions>`, that'll let the TS compiler know you don't intend on using the entire interface, but just a part of it.

**Comment by anselanza** [comment]
I'm afraid that doesn't do it, either, because the `oscillator` property (which in turn is defined by `OmitSourceOptions<ToneCustomOscillatorOptions>`) still isn't satisfied as far as the compiler is concerned.
<img width="393" alt="Screenshot 2021-11-02 at 10 13 07" src="https://user-images.githubusercontent.com/3015222/139818612-8edb17c5-f19f-45c6-af5d-0e36f0fefb99.png">

Sorry if this is tedious, and feels like a "debugging session via Issue", but I am genuinely curious what the correct/intended approach should be here. It seems like a rather common use case to want to define some options for a synth/instrument and pass these as arguments or props while enjoying the convenience of the type system checking for the required properties (and preferably _only_ the required properties).

Is there a problem with the way the type definitions are currently defined in the library? Or do we need some clear examples that show the proper way to use the type definitions in a case such as this?
